### ISSUE TITLE
OpenAPI add-on Import should honor default: for PostBody params, like it does for Header params

### ISSUE THREAD

**Issue by davewichers** [zaproxy/zaproxy#6095]
I've tested 2 OpenAPI scenarios:

This one is honored:
```
  /hash-00/BenchmarkTest00009:
    post:
      operationId: doheader_9
      parameters:
      - in: header
        name: whatever
        required: true
        schema:
          type: string
          default: BenchmarkTest00009  <-- This default value IS included as the 'whatever' header value in the call ZAP makes for this endpoint
      responses:
        default:
          description: default response
          content:
            text/html: {}
```

While this one is not:
```
  /crypto-00/BenchmarkTest00035:
    post:
      operationId: dopostNames_35
      requestBody:
        description: Form POST parameters in request body
        required: true
        content:
          application/x-www-form-urlencoded:
            schema:
              type: object
              properties:
                answer:
                  type: string
                whatever:
                  type: string
                  default: BenchmarkTest00035  <-- This default value is ignored
```

ZAP simply includes whatever=whatever as the name/value of this post body parameter.

**Describe the solution you'd like**
I'd like to see the initial call generated be:  whatever=BenchmarkTest00035

As specified by the default, like the header parameter does.

**Comment by psiinon** [comment]
Sounds very reasonable.

**Comment by davewichers** [comment]
@thc202 - I just downloaded the latest ZAP weekly and tested it to see if the fix implemented in zaproxy/zap-extensions#2542 worked, and I didn't see any change in behavior. I'm using ZAP D-2020-11-18 and it says it is using the OpenAPI Support plugin v17.0.0. Has the fix implemented in: zaproxy/zap-extensions#2542 been pushed to a new release of the OpenAPI plugin? If not, when do you expect that to happen?

**Comment by kingthorin** [comment]
It hasn't been released yet: https://github.com/zaproxy/zap-extensions/blob/master/addOns/openapi/CHANGELOG.md

**Comment by thc202** [comment]
It should be included in the weekly, could you provide an example not working?

**Comment by davewichers** [comment]
I tested both fixes against the exact OWASP Benchmark test cases referenced
in each ticket as examples of it not (originally) working.  You can test
them yourself by running Benchmark, and using the openapi spec file for
Benchmark that is in the data/ folder.

On a related note, if the ZAP Weekly says it is using version 17.0.0 of the
OpenAPI plugin, then why would you think it should be using the 'latest'
version of that plugin from the plugins trunk? If the ZAP weekly was using
weekly builds of the plugins, wouldn't they have different version numbers
than the released versions? Something like 18.0.0-SNAPSHOT or something?
i.e., shouldn't it be easy to tell exactly which version of the plugin ZAP
is using if it isn't using released versions?

-Dave

**Comment by thc202** [comment]
I'd suggest cleaning up the ZAP dev home or use a new one. It's most likely picking an older version without the changes.

Historical reasons, the add-ons don't use the SNAPSHOT suffix.

**Comment by davewichers** [comment]
OK. I'll give that a shot. What version does the development version of a
plugin use? The next #? i.e., if I have things configured right, should I
be seeing version 18.0.0 of the OpenAPI plugin being used by the ZAP
weekly, which is the next available number beyond 17.0.0 which is the
current public release?

**Comment by thc202** [comment]
Usually it uses the next number, It should be version 17 as well. The current version available in the marketplace is 16.

**Comment by davewichers** [comment]
How exactly do I "clean up the ZAP dev home or use a new one."?  Do something within ZAP? Do something at the command line before starting ZAP? Again, I'm using a weekly release.

**Comment by thc202** [comment]
For the location where the default home is: https://www.zaproxy.org/faq/what-is-the-default-directory-that-zap-uses/ (See bottom of the page for weekly release note.)

In this case you can just delete whatever is in the `plugin` directory. You can start ZAP with `-dir` command line argument to use a new home. https://www.zaproxy.org/docs/desktop/cmdline/

**Comment by davewichers** [comment]
I found this very confusing, trying to figure out which plugin was working, and how to find something, delete it, etc. so the weekly release started using the included plugins (which seems to me should be the default). Is there a way to improve this to make it more likely to 'just work'? Should I open a separate ticket for this request?

Also, I was able to test the fix for the default post body parameter value, and it does seem to work as requested. Thx
